### PR TITLE
[DOCS] Update docs

### PR DESCRIPTION
**Pull request by alexander-nitsche**
Hi Oliver, hi Tea team,

this PR is a suggestion to align your extension with the recently introduced [TYPO3 documentation standards](https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/GeneralConventions/FileStructure.html).

I transformed your existing rather single file documentation into a full documentation, and split the content into several files. I further reworked some content, most prominently added the full list of composer commands, so that they can be found via the global search at docs.typo3.org.

Greetings, Alex

Related: https://github.com/TYPO3-Documentation/T3DocTeam/issues/181

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,239 +1,22 @@
-# Example TYPO3 extension for code quality checks and automated tests
```
Comment (oliverklee) [inline README.md:1]: Please let's keep the main heading the first thing in the document (above the badges).

File: README.md [hunk 2]
```diff
@@ -1,239 +1,22 @@
...
-
-[![GitHub CI status](https://github.com/TYPO3-Documentation/tea/workflows/CI/badge.svg?branch=main)](https://github.com/TYPO3-Documentation/tea/actions)
-[![GitLab CI status](https://gitlab.typo3.org/qa/example-extension/badges/main/pipeline.svg)](https://gitlab.typo3.org/qa/example-extension/-/pipelines)
 [![Latest stable version](https://poser.pugx.org/ttn/tea/v/stable.svg)](https://packagist.org/packages/ttn/tea)
-[![TYPO3](https://img.shields.io/badge/TYPO3-10-orange.svg)](https://get.typo3.org/version/10)
+[![Total downloads](https://poser.pugx.org/ttn/tea/downloads.svg)](https://packagist.org/packages/ttn/tea)
 [![TYPO3](https://img.shields.io/badge/TYPO3-11-orange.svg)](https://get.typo3.org/version)
+[![TYPO3](https://img.shields.io/badge/TYPO3-10-orange.svg)](https://get.typo3.org/version/10)
```
Comment (oliverklee) [inline README.md:4]: Please let's keep the current order of the version badges: from lowest to highest.

File: README.md [hunk 3]
```diff
@@ -1,239 +1,22 @@
...
+[![GitHub CI status](https://github.com/TYPO3-Documentation/tea/workflows/CI/badge.svg?branch=main)](https://github.com/TYPO3-Documentation/tea/actions)
+[![GitLab CI status](https://gitlab.typo3.org/qa/example-extension/badges/main/pipeline.svg)](https://gitlab.typo3.org/qa/example-extension/-/pipelines)
 [![Code Coverage Badge](https://github.com/TYPO3-Documentation/tea/blob/code-coverage-badge/badge.svg)](https://github.com/TYPO3-Documentation/tea/blob/code-coverage-badge/clover.xml)
-[![Total downloads](https://poser.pugx.org/ttn/tea/downloads.svg)](https://packagist.org/packages/ttn/tea)
-[![License](https://poser.pugx.org/ttn/tea/license.svg)](https://packagist.org/packages/ttn/tea)
```
Comment (oliverklee) [inline README.md:10]: Please don't remove the license badge.

**Review by oliverklee — CHANGES_REQUESTED** [review]
Thanks for this monumental contribution, @alexander-nitsche! :heart:

As the first review pass, I have added my remarks concerning the documentation structure.

**Comment by oliverklee** [comment]
@alexander-nitsche Do you have time to update the PR during the next few days, or should someone from the team take over?

**Comment by alexander-nitsche** [comment]
hi @oliverklee , just go ahead and adapt the PR to your likes. greetings, alex

**Comment by alexander-nitsche** [comment]
Added two new commits regarding badges in README and additional sources (TER + docs.typo3.org) in composer.json.

In addition, the tea [TER homepage](https://extensions.typo3.org/extension/tea) is missing the "Extension Manual" and "Packagist" buttons, which seem to be caused by an empty "Composer name of extension" field in the TER extension edit form.

File: Documentation/Index.rst [hunk 4]
```diff
@@ -1,63 +1,60 @@
...
 
-The content of this document is related to TYPO3 CMS,
-a GNU/GPL CMS/Framework available from `typo3.org <https://typo3.org/>`_ .
+This TYPO3 extension, based on Extbase and Fluid, is an example of best
+practices in automated code checks, unit/functional/acceptance testing and
+continuous integration.
 
-For Contributors
-================
+----
```
[the remark this answers is not in the document]
Comment (alexander-nitsche) [reply Documentation/Index.rst:37]: It's a horizontal line that just looks good, see e.g. https://docs.typo3.org/p/friendsoftypo3/tt-address/main/en-us/.

File: README.md (hunk 2 above)
Comment (oliverklee) [reply README.md:5]: Done.

File: README.md (hunk 3 above)
Comment (oliverklee) [reply README.md:10]: Done.

File: README.md [hunk 5]
```diff
@@ -1,239 +1,22 @@
...
 [![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.0-4baaaa.svg)](CODE_OF_CONDUCT.md)
 
-## What is this all about?
+# TYPO3 extension `tea`
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply README.md:11]: Done.

File: Documentation/ContinuousIntegration/Index.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
+.. include:: /Includes.rst.txt
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply Documentation/ContinuousIntegration/Index.rst]: Done.

**Review by oliverklee — APPROVED** [review]
Thanks, Alexander! :heart:

**Pull request merged**